pkg/ibmvpc/vpc.go

```diff
@@ -592,34 +592,34 @@ type SecurityGroup struct {
 }
 
 func (sg *SecurityGroup) AllowedConnectivity(src, dst vpcmodel.Node, isIngress bool) *connection.Set {
-	memberStrAddress, targetIPBlock := sg.getMemberTargetStrAddress(src, dst, isIngress)
-	if _, ok := sg.members[memberStrAddress]; !ok {
+	memberIPBlock, targetIPBlock := sg.getMemberTargetStrAddress(src, dst, isIngress)
+	if _, ok := sg.members[memberIPBlock.ToIPAddressString()]; !ok {
 		return connection.None() // connectivity not affected by this SG resource - input node is not its member
 	}
-	return sg.analyzer.AllowedConnectivity(targetIPBlock, isIngress)
+	return sg.analyzer.AllowedConnectivity(targetIPBlock, memberIPBlock, isIngress)
 }
 
 // rulesFilterInConnectivity list of SG rules contributing to the connectivity
 func (sg *SecurityGroup) rulesFilterInConnectivity(src, dst vpcmodel.Node, conn *connection.Set,
 	isIngress bool) (tableRelevant bool, rules []int, err error) {
-	memberStrAddress, targetIPBlock := sg.getMemberTargetStrAddress(src, dst, isIngress)
-	if _, ok := sg.members[memberStrAddress]; !ok {
+	memberIPBlock, targetIPBlock := sg.getMemberTargetStrAddress(src, dst, isIngress)
+	if _, ok := sg.members[memberIPBlock.ToIPAddressString()]; !ok {
 		return false, nil, nil // connectivity not affected by this SG resource - input node is not its member
 	}
-	rules, err = sg.analyzer.rulesFilterInConnectivity(targetIPBlock, conn, isIngress)
+	rules, err = sg.analyzer.rulesFilterInConnectivity(targetIPBlock, memberIPBlock, conn, isIngress)
 	return true, rules, err
 }
 
 func (sg *SecurityGroup) getMemberTargetStrAddress(src, dst vpcmodel.Node,
-	isIngress bool) (memberStrAddress string, targetIPBlock *ipblock.IPBlock) {
+	isIngress bool) (memberStrAddress, targetIPBlock *ipblock.IPBlock) {
 	var member, target vpcmodel.Node
 	if isIngress {
 		member, target = dst, src
 	} else {
 		member, target = src, dst
 	}
 	// TODO: member is expected to be internal node (validate?) [could use member.(vpcmodel.InternalNodeIntf).Address()]
-	return member.CidrOrAddress(), target.IPBlock()
+	return member.IPBlock(), target.IPBlock()
```

Comment:
- add a third return value that preserves the original `member.CidrOrAddress()` value returned